src/main/java/org/opensearch/ad/transport/BackPressureRouting.java

```diff
@@ -45,18 +46,23 @@
     private static final Logger LOG = LogManager.getLogger(BackPressureRouting.class);
     private final String nodeId;
     private final Clock clock;
-    private final int maxRetryForUnresponsiveNode;
+    private volatile int maxRetryForUnresponsiveNode;
     private final TimeValue mutePeriod;
     private AtomicInteger backpressureCounter;
     private long lastMuteTime;
+    private ClusterService clusterService;
 
-    public BackPressureRouting(String nodeId, Clock clock, Settings settings) {
+    public BackPressureRouting(String nodeId, Clock clock, Settings settings, ClusterService clusterService) {
         this.nodeId = nodeId;
         this.clock = clock;
         this.backpressureCounter = new AtomicInteger(0);
         this.maxRetryForUnresponsiveNode = MAX_RETRY_FOR_UNRESPONSIVE_NODE.get(settings);
         this.mutePeriod = BACKOFF_MINUTES.get(settings);
         this.lastMuteTime = 0;
+        this.clusterService = clusterService;
```

Comment:
Have you tried to verify dynamically changing max retry for unresponsive node works?  You need to simulate a node is not responsive by changing the code to reject every incoming request and check if the node will be filtered out after the set unresponsive times?

Reply:
This applies to all settings you have changed.